pkg/virt-api/webhooks/validating-webhook/admitters/vmclone-admitter.go

```diff
@@ -411,3 +416,29 @@ func validateCloneVolumeSnapshotSupportVMSnapshotContent(snapshotContents *v1alp
 
 	return result
 }
+
+func validateNewMacAddresses(vmClone *clonev1alpha1.VirtualMachineClone) []metav1.StatusCause {
```

Comment:
I think we can reduce code duplication by changing the original `validateMacAddress` function to something like this:
```
func validateMacAddress(ifaceName, ifaceFieldPath, macAddress string) (causes []metav1.StatusCause) {
```
where `ifaceName`, `ifaceFieldPath`, and `macAddress` are passed by the caller instead of being calculated inside the function itself.

For the original vmi-create-admitter for example the caller should be changed to:
```
		causes = append(causes, validateMacAddress(
			field.Child("domain", "devices", "interfaces").Index(idx).Child("name").String(),
			field.Child("domain", "devices", "interfaces").Index(idx).Child("macAddress").String(),
			iface.MacAddress)...)
```

WDYT? @alicefr feedback?

Reply:
@fossedihelm sounds like a great approach!

Reply:
If we do this, do we need to move the `func validateMacAddress()` out of vmi-create-admitter.go and into a new file? what do you think? If possible, what should we name this file?

Reply:
@fossedihelm  Do you think the `validateMacAddress()`functions in `vmi-create-admitter.go`can be referenced in `vmclone-admitter.go`?

Reply:
@matthewei It will deserve a refactoring for sure, but right now yes, they are in the same package

Reply:
I see, but in this PR, Do you think it is necessary to refactor this function?

Reply:
I don't think it is worth it to do it here. In a follow-up PR is enough :)

Reply:
I see, will do it

Reply:
done @fossedihelm